src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/CohortController.cs

```diff
@@ -168,9 +173,32 @@ public IActionResult AddDraftApprenticeship2(AddDraftApprenticeshipViewModel mod
         [Authorize(Policy = nameof(PolicyNames.HasContributorOrAbovePermission))]
         public async Task<IActionResult> SelectDeliveryModel(AddDraftApprenticeshipViewModel request)
         {
-            return View("SelectDeliveryModel", request);
+            var models = (await GetProviderCourseDeliveryModels(request.ProviderId, request.CourseCode)).ToArray();
+
+            if (models.Count() > 1)
+            {
+                return RedirectToAction("SelectDeliveryModel", request);
+            }
+
+            request.DeliveryModel = models.FirstOrDefault();
+            return RedirectToAction("AddDraftApprenticeship", request);
         }
 
+        [HttpPost]
+        [Route("select-delivery-model")]
+        [Authorize(Policy = nameof(PolicyNames.HasContributorOrAbovePermission))]
+        public async Task<IActionResult> SetDeliveryModel(AddDraftApprenticeshipViewModel request)
+        {
+            if (request.DeliveryModel == null)
+            {
+                throw new CommitmentsApiModelException(new List<ErrorDetail>
+                    {new ErrorDetail("DeliveryModel", "Please select a delivery model option")});
+            }
+
+            return View("AddDraftApprenticeship", request);
```

Comment:
This is probably a question for the next round of work - doesn't returning `View(...)` here mean that the URL will remain as `select-delivery-model` even though the page displayed is actually the draft apprenticeship details?

Reply:
yes should be redirect